source/ice_api.c

```diff
@@ -0,0 +1,1835 @@
+                                            newIpAddr.isPointToPoint = 0;
+
+                                            retStatus = Ice_CheckPeerReflexiveCandidate( pIceAgent,
+                                                                                         &( newIpAddr ),
+                                                                                         pLocalCandidate,
+                                                                                         pIceCandidatePair->pLocal->priority,
+                                                                                         false );
+
+                                            if( retStatus == ICE_RESULT_OK )
+                                            {
+                                                stunPacketHandleStatus = ICE_RESULT_FOUND_PEER_REFLEXIVE_CANDIDATE;
+                                            }
+                                        }
+                                    }
+                                    else
+                                    {
+                                        /* No mapped address attribute found in STUN response. Dropping Packet. */
+                                        stunPacketHandleStatus = ICE_RESULT_NOT_FOUND_ADDRESS_ATTRIBUTE;
+                                    }
+                                }
+                            }
+                            else
+                            {
+                                /* Dropping response packet because transaction id does not match.*/
+                                stunPacketHandleStatus = ICE_RESULT_NOT_FOUND_MATCHING_TRANSACTION_ID;
+                            }
+                        }
+                        else
+                        {
+                            /* Candidate Pair was not found using the local and remote IP address. */
+                            stunPacketHandleStatus = ICE_RESULT_NOT_FOUND_CANDIDATE_PAIR;
+                        }
+                    }
+                }
+
+                break;
+
+            case STUN_MESSAGE_TYPE_BINDING_INDICATION:
+                /* Received STUN binding indication */
+                stunPacketHandleStatus = ICE_RESULT_STUN_BINDING_INDICATION;
+                break;
+
+            default:
+                /* STUN packet received is neither a Binding Request nor a Response. */
+                stunPacketHandleStatus = ICE_RESULT_STUN_INVALID_PACKET_TYPE;
+                break;
+        }
+    }
+
+    if( foundCandidatePair )
+    {
+        *ppIceCandidatePair = pIceCandidatePair;
+    }
+
+    return stunPacketHandleStatus;
+}
+
+/*------------------------------------------------------------------------------------------------------------------*/
+
+/* Ice_GetValidLocalCandidateCount - Get valid Local Candidate count */
+
+int Ice_GetValidLocalCandidateCount( IceAgent_t * pIceAgent )
+{
+    int i = -1;
```

Comment:
Can we name the return value to a macro called NoValidLocalCandidate for the case where no candidates are found. This also helps the application to check against this

Reply:
updated.